addons/osio-addon/src/main/java/io/fabric8/launcher/osio/identity/OsioIdentityProvider.java

```diff
@@ -2,37 +2,38 @@
 
 import java.util.Optional;
 
-import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
 
 import io.fabric8.launcher.base.identity.Identity;
 import io.fabric8.launcher.base.identity.IdentityFactory;
-import io.fabric8.launcher.base.identity.TokenIdentity;
 import io.fabric8.launcher.core.spi.Application;
 import io.fabric8.launcher.core.spi.IdentityProvider;
 import io.fabric8.launcher.osio.client.OsioApiClient;
-import io.fabric8.launcher.osio.client.OsioApiClientImpl;
+import io.fabric8.launcher.osio.client.Tenant;
 
-import static io.fabric8.launcher.base.identity.IdentityFactory.createFromToken;
-import static io.fabric8.launcher.base.identity.IdentityHelper.removeBearerPrefix;
 import static io.fabric8.launcher.core.spi.Application.ApplicationType.OSIO;
 import static io.fabric8.launcher.osio.OsioConfigs.ExternalServices.getGithubServiceName;
 
 
-@ApplicationScoped
+@Dependent
```

Comment:
Afaik Using @Dependent will make the bean reuse the scope of the consumer. If you are injecting it in an @ApplicationScoped endpoint (like ValidationResource) it will assume the parent scope.

Reply:
I changed it in ValidationResource by placing it in the method params ;)